test/integration/volume/attach_detach_test.go

```diff
@@ -446,3 +523,85 @@ func TestPodAddedByDswp(t *testing.T) {
 
 	close(stopCh)
 }
+
+func TestPVCBoundWithADC(t *testing.T) {
+	_, server, closeFn := framework.RunAMaster(framework.NewIntegrationTestMasterConfig())
+	defer closeFn()
+	namespaceName := "test-pod-deletion"
+
+	ns := framework.CreateTestingNamespace(namespaceName, server, t)
+	defer framework.DeleteTestingNamespace(ns, server, t)
+
+	testClient, ctrl, pvCtrl, informers := createAdClients(ns, t, server, defaultSyncPeriod, attachdetach.TimerConfig{
+		ReconcilerLoopPeriod:                        100 * time.Millisecond,
+		ReconcilerMaxWaitForUnmountDuration:         6 * time.Second,
+		DesiredStateOfWorldPopulatorLoopSleepPeriod: 24 * time.Hour,
+		// Use high duration to disable DesiredStateOfWorldPopulator.findAndAddActivePods loop in test.
+		DesiredStateOfWorldPopulatorListPodsRetryDuration: 24 * time.Hour,
+	})
+
+	node := &v1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "node-sandbox",
+			Annotations: map[string]string{
+				util.ControllerManagedAttachAnnotation: "true",
+			},
+		},
+	}
+	if _, err := testClient.Core().Nodes().Create(node); err != nil {
+		t.Fatalf("Failed to created node : %v", err)
+	}
+
+	// pod with pvc not bound
+	pod, pvc := fakePodWithPVC(namespaceName)
+	if _, err := testClient.Core().Pods(pod.Namespace).Create(pod); err != nil {
+		t.Errorf("Failed to create pod : %v", err)
+	}
+	if _, err := testClient.Core().PersistentVolumeClaims(pvc.Namespace).Create(pvc); err != nil {
+		t.Errorf("Failed to create pvc : %v", err)
+	}
+	// pod with no pvc
+	podNew := fakePodWithVol(namespaceName)
+	podNew.SetName(podNew.Name + "other")
+	if _, err := testClient.Core().Pods(podNew.Namespace).Create(podNew); err != nil {
+		t.Errorf("Failed to create pod : %v", err)
+	}
+
+	// start controller loop
+	stopCh := make(chan struct{})
+	informers.Start(stopCh)
+	informers.WaitForCacheSync(stopCh)
+	go ctrl.Run(stopCh)
+	go pvCtrl.Run(stopCh)
+
+	waitToObservePods(t, informers.Core().V1().Pods().Informer(), 2)
+	// Give attachdetach controller enough time to populate pods into DSWP.
+	time.Sleep(10 * time.Second)
+	waitForPodFuncInDSWP(t, ctrl.GetDesiredStateOfWorld(), 10*time.Second, "expected 1 pod in dsw", 1)
+	createPVForPVC(t, testClient, pvc)
+	waitForPodFuncInDSWP(t, ctrl.GetDesiredStateOfWorld(), 20*time.Second, "expected 2 pods in dsw after PVCs are bound", 2)
+	close(stopCh)
```

Comment:
I wonder how well 10s and 20s setting will work? What is the typical delay under normal load, and heavy load?

Reply:
attachdetach controller processes Pod's PVs to attach very fast, the delay should be almost same as apiserver watch event delay, because it does nothing except adding Pod's Volumes into DSWP (no api calls) on event. 
In my local test, delay to attach PV is less than 100ms. The most delay is waiting PV controller to bound PVC, normally 4~5 seconds.
Perhaps we should increase `timeout` of `waitForPodFuncInDSWP`, and it will not increase execution time of test.
`time.Sleep(10 * time.Second)` is a hack to make sure the second pod cannot be added into DSWP at least in given seconds, because no object state or event can be checked to verify. 10 seconds should be enough here.

Reply:
`timeout` of `waitForPodFuncInDSWP` is increased to 60 seconds now.